src/RestApi/StoreApi/Routes/Checkout.php

```diff
@@ -164,7 +326,7 @@ protected function process_payment( \WC_Order $order, \WP_REST_Request $request
 			 * @param PaymentContext $context Holds context for the payment, including order ID and payment method.
 			 * @param PaymentResult  $result Result object for the transaction.
 			 */
-			do_action( 'woocommerce_rest_checkout_process_payment_with_context', $context, $result );
+			do_action_ref_array( 'woocommerce_rest_checkout_process_payment_with_context', [ $context, &$result ] );
```

Comment:
It shouldn't be necessary to pass `PaymentResult` as a reference value since in PHP all objects are automatically references anyways right?

Reply:
That is what I thought but I spotted some notices in my error log which went away with this change. WP core does this when it explicitly needs references.

Reply:
Odd, and concerning.  I assumed WP core just has it because of legacy code.